### ISSUE TITLE
Add this to awesome-graphql

### ISSUE THREAD

**Issue by tomprince** [jml/graphql-api#110]
This probably wants to be added to https://github.com/chentsulin/awesome-graphql#lib-haskell

**Comment by teh** [comment]
Thanks for the pointer! 

I think I'd like to maybe wait out one more release while we work out the roughest edges. E.g. today I realized that our tests don't work when downloading from hackage (#109). I'm certain we have few more shockers like that :)

**Comment by tomprince** [comment]
Yeah, I thought that might be the case, which is why I didn't submit it my self.

**Comment by jasonzoladz** [comment]
I will literally pay you $250 if you get this into production shape by the end of March.  :)  Thanks for working on this.  (Aside:  It makes me a little sad that Elixir has two full-featured graphql implementations and we Haskellers don't yet have one.)

Btw, I put together a fix for the bounds checking on the ValueInt parser here: https://github.com/jdnavarro/graphql-haskell/blob/master/Data/GraphQL/Parser.hs#L148-L168

You might consider incorporating something similar.

**Comment by teh** [comment]
@jasonzoladz please raise specific issues for changes or fixes. I'm currently implementing a project using this library, and I'm making a list of what I want changed. The more use cases we see the stronger the library will be!

If you want a production ready library by end of March I think you're looking at 50x-100x that amount :stuck_out_tongue_winking_eye:

**Comment by jasonzoladz** [comment]
Haven't had any issues with the library so far.  Well done.

Questions:

* Have you thought about adding something akin to Servant's `AuthProtect`?
* Would you consider posting a list of todos for introspection support?  I'd like to help if I can.

**Comment by jml** [comment]
Thanks @jasonzoladz 

> Have you thought about adding something akin to Servant's AuthProtect?

Although I'm not intimately familiar with `AuthProtect`, we _have_ thought about adding authentication support, and would rather not just yet. Our hope is that this can be managed at a separate layer to everyone's satisfaction. What are you trying to do?

> Would you consider posting a list of todos for introspection support? I'd like to help if I can.

Thanks for the offer! I think @teh might have done some thinking on this. As for me, I'd need to sit and figure things out, which would take some time and end up producing code as an artefact.

**Comment by jasonzoladz** [comment]
Presently, when doing auth, each `graphql-api` handler that requires auth needs an `Either ErrMsg User` as its first argument.  It might be nice to be able to factor that out and just have sub-handlers underneath a single authentication combinator -- _a la Servant_.  However, it's not nearly as big a deal as introspection.
